server/index.ts

```diff
@@ -226,6 +206,61 @@ const server = async ({ port = 3000 }) => {
             onSuccess?.(players);
         });
 
+        socket.on(USER_EVENTS.READY_FOR_NEXT_TRICK, async (gameId: string, playerId: string, onSuccess, onError) => {
+            try {
+                const [roundNumber, trickNumber, roundComplete, trickComplete] = await Promise.all([
+                    db.getCurrentRound(redis, gameId),
+                    db.getCurrentTrick(redis, gameId),
+                    db.currentRoundIsComplete(redis, gameId),
+                    db.currentTrickIsComplete(redis, gameId),
+                ]);
+
+                if (!trickComplete) {
+                    throw new Error(`Player (${playerId}) cannot say ready before the current trick (${trickNumber}) is complete`)
+                }
+                await db.setPlayerReady(redis, gameId, roundNumber, trickNumber, playerId);
+                const [players, playersReady,] = await Promise.all([
+                    db.getGamePlayers(redis, gameId),
+                    db.getPlayersReady(redis, gameId, roundNumber, trickNumber),
+                ]);
+
+                onSuccess?.();
+
+                io.to(gameId).emit(SERVER_EVENTS.PLAYER_READY, playerId);
+
+                if (players.length === playersReady.length) {
+                    if (!roundComplete) {
+                        await db.setCurrentTrick(redis, gameId, trickNumber + 1);
+                    } else {
+                        await Promise.all([
+                            db.setCurrentTrick(redis, gameId, 0),
+                            db.setCurrentRound(redis, gameId, roundNumber + 1),
+                        ]);
+                    }
+
+                    if (roundComplete) {
```

Comment:
why not combine this with the if/else above?

Reply:
sure i can do that.  I just moved if from the event above is why it was left like this